tests/integration/__init__.py

```diff
@@ -108,6 +155,38 @@ def cleanup_runtime_config_instance(to_cleanup):
 
 atexit.register(cleanup_runtime_config_instance, RUNTIME_CONFIGS)
 
+_RUNTESTS_PORTS = {}
+
+
+def get_unused_localhost_port():
+    '''
+    Return a random unused port on localhost
+    '''
+    usock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM)
+    usock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    usock.bind(('127.0.0.1', 0))
+    port = usock.getsockname()[1]
+    if port in (54505, 54506, 64505, 64506, 64510, 64511):
+        # These ports are hardcoded in the test configuration
+        usock.close()
+        return get_unused_localhost_port()
```

Comment:
Well salt should also not start if those ports are already bind, so the above change would also require that we verify port binding before starting and bailout if we can't.

A good approach but it touches a lot in salt. That could probably be addressed in another pull request...